### PR TITLE
Add initial development setup

### PR DESCRIPTION
**Pull request by jpellizzari**
Adds the initial baseline for dev work to begin on the UI.

While the exact means of distribution for this GUI is still not clear, I assume we will want to embed assets inside a `go` binary at some point, so this PR has an asset generation -> embedding flow that can be used for that. There is some build tag trickery to generate the embed-able assets using a package called [vfsgen](github.com/shurcooL/vfsgen).
 
Important points about some of the other choices:
* Uses `parcel` as the front-end bundler. This is a superior choice to `webpack` in my opinion, as it requires much less configuration
* Uses a development middleware server to proxy requests to the backend and do all the fancy React hot-reloading you would expect
* React stateless-functional-components and hooks will be used for state management (as opposed to `redux`). Much of the advantages of `redux` can be better encapsulated in a hook.

### CODE REVIEW

File: Makefile
Comment (squaremo) [inline Makefile:10]: Minor nit: `build` or `generate` might be a more accurate tag to use here -- `dev` is usually reserved for things that you only run locally (i.e., while trying out new code).

**Review by squaremo — APPROVED** [review]
The bits I can claim to grok (i.e., not the JS framework bits) look like a good start to me :+1: (NB remember you'll need another maintainer -- i.e, @bia -- to approve, before merging.)

**Review by bia — APPROVED** [review]
WFM 🎉

File: README.md
Comment (hiddeco) [inline README.md:11]: Technically a requirement, as it is not installed by default like `npm`? Also, not really sure `yarn` still offers many advantages over [`npm` v7](https://github.blog/2020-10-13-presenting-v7-0-0-of-the-npm-cli/).

Additional note from the original author: https://twitter.com/sebmck/status/1300655408688709633
Additional alternative: https://pnpm.js.org/en/motivation

**Comment by jpellizzari** [comment]
@stefanprodan  It seems that the 800+ dependencies come from the [flux/pkg/runtime](https://github.com/fluxcd/pkg/blob/main/runtime/go.sum) and [prometheus/client_golang](https://github.com/prometheus/client_golang/blob/master/go.sum) modules :man_shrugging:.

The code generation library is behind a build tag, so I don't think it gets included?

File: README.md
Comment (jpellizzari) [reply README.md]: Removed `yarn` as a dependency and updated the docs.

File: Makefile
Comment (jpellizzari) [reply Makefile]: Fixed

**Review by hiddeco — APPROVED** [review]
Thank you for kicking this off @jpellizzari :1st_place_medal:

**Review by stefanprodan — APPROVED** [review]

**Pull request merged**
